### PR TITLE
Harmonise versions with StrIoT 

### PR DESCRIPTION
**Pull request by JonnySpruce**
Change resolver, match package versions with StrIoT, and reintroduce `package.yaml`

~~This was unfortunately not as successful as I had hoped, since using resolver `lts-13.25` (same as StrIoT) led to an impossible situation where the `network` package was required at both `<2.9` AND `>3.1` - therefore with no compatible overlap. @jmtd @adam-cattermole - I don't know if there is anything I can do to resolve this and  still use `lts-13.25`?~~

Changes resolver to `lts-13.25` and changes all package versions to match the StrIoT versions. 

As part of this PR I also:
- updated the licence to Apache.
- tidied up the `stack.yaml` and reintroduced the `package.yaml` to help with the building of the `.cabal` file.

### CODE REVIEW

**Comment by adam-cattermole** [comment]
Hi @JonnySpruce,

Unfortunately the version of the `network` package we use is not in any of the stack snapshots until `lts-15.0`. To get this working you have to include all the new versions of the dependencies; `socks`, `websockets` and `connection`. I can get it all building on `lts-13.25` with the following suggested changes.

File: stack.yaml [hunk 1]
```diff
@@ -4,15 +4,12 @@ packages:
 extra-deps:
   - git: https://github.com/striot/striot.git
     commit: 1441100588615723481b9d7571dbb5b37bbd65d4
-  - algebraic-graphs-0.5
   - aeson-1.4.6.0
```
Comment (adam-cattermole) [inline stack.yaml:7]:
```suggestion
  - aeson-1.4.6.0
  - network-3.1.0.0
  - unagi-chan-0.4.1.0
  - time-compat-1.9.2.2
  - websockets-0.12.6.1
  - connection-0.3.1
  - socks-0.6.1
```
If you add these extra dependencies and change the snapshot to `lts-13.25` it all seems to build!

File: striot-gui.cabal [hunk 2]
```diff
@@ -69,10 +67,9 @@ test-suite striot-gui-test
   build-depends:
       aeson >=1.4.6.0
     , algebraic-graphs >=0.1
-    , base >=4.7 && <5
-    , bytestring >=0.10.10.0
+    , base >=4.9
+    , bytestring >=0.9.2
+    , hspec >=2.7.1
```
Comment (adam-cattermole) [inline striot-gui.cabal]:
```suggestion
    , hspec >=2.6.1
```
If you don't need the extra changes in version `2.7.1` this version is included in `lts-13.25`.

File: stack.yaml [hunk 3]
```diff
@@ -4,15 +4,12 @@ packages:
...
-  - bytestring-0.10.10.0
   - hspec-2.7.1
```
Comment (adam-cattermole) [inline stack.yaml]:
```suggestion
```
You can remove this line here if you opt for version `2.6.1`

**Comment by JonnySpruce** [comment]
Thanks @adam-cattermole, it seems like that did the trick!

**Comment by adam-cattermole** [comment]
No problem at all - need any help with anything else just give me a shout!

**Pull request merged**